src/lasp_dependence_dag.erl

```diff
@@ -0,0 +1,202 @@
+-module(lasp_dependence_dag).
+
+-include("lasp.hrl").
+-behaviour(gen_server).
+
+%% API
+-export([start_link/0,
+         is_loop/2,
+         add_edges/3,
+         add_vertex/1,
+         add_vertices/1]).
+
+%% Test
+-export([n_vertices/0,
+         n_edges/0,
+         out_degree/1,
+         in_degree/1,
+         out_edges/1,
+         in_edges/1]).
+
+%% gen_server callbacks
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3]).
+
+%%%===================================================================
+%%% Type definitions
+%%%===================================================================
+
+-record(state, {dag :: digraph:graph()}).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+-spec add_vertex(id()) -> ok.
+add_vertex(V) ->
+    gen_server:call(?MODULE, {add_vertex, V}, infinity).
+
+-spec add_vertices(list(id())) -> ok.
+add_vertices([]) ->
+    ok;
+
+add_vertices(Vs) ->
+    gen_server:call(?MODULE, {add_vertices, Vs}, infinity).
+
+%% @doc Check if linking the given vertices will form a loop.
+-spec is_loop(list(id()), id()) -> boolean().
+is_loop(Src, Dst) ->
```

Comment:
OK, can we rename the loop terminology to be closer to this to be clearer in the code before we merge?